lib/Interpreter/SQL/MySQLAggregateOptimizer.php

```diff
@@ -78,18 +78,22 @@ public function __construct(Interpreter\Interpreter $interpreter) {
 	private function validateAggregationUsage() {
 		if ($this->aggValid === null) {
 			$this->aggValid = false;
+			$levels = Util\Aggregation::getAggregationLevels();
+			$level = $this->groupBy ? Util\Aggregation::getAggregationLevelTypeValue($this->groupBy) : -1;
+			while ($level >= 0 && !$this->aggValid) {
```

Comment:
Please add a comment:

    // iterate Aggregation::$aggregationLevels in reverse order, starting with highest possible level ($groupBy)

Reply:
I'm still feeling this could be done more elegantly using array_key, array_reverse, array_search but I couldn't come up with anything, so fine with me.

Reply:
resolved by 0106b52